### PR TITLE
Implement ensureCompanyFunding

### PR DESCRIPTION
**Pull request by duncanspumpkin**
Not 100% on the name

### CODE REVIEW

File: src/OpenLoco/src/World/CompanyManager.cpp [hunk 1]
```diff
@@ -617,6 +617,67 @@ namespace OpenLoco::CompanyManager
...
+                while (company->cash + additionalFunds < payment)
+                {
+                    if (company->currentLoan + additionalFunds > maxLoan)
+                    {
+                        FormatArguments::common(payment);
+                        GameCommands::setErrorText(StringIds::not_enough_cash_requires_currency32);
+                        return false;
+                    }
+                    additionalFunds += 1000;
+                }
```
Comment (ZehMatt) [inline src/OpenLoco/src/World/CompanyManager.cpp:671]: Do we really need the loop here? I think we can just calculate the difference and align it back to 1000 and see if its more than its allowed.

File: src/OpenLoco/src/World/CompanyManager.cpp [hunk 2]
```diff
@@ -617,6 +617,67 @@ namespace OpenLoco::CompanyManager
         company->expenditures[0][static_cast<uint8_t>(type)] -= payment;
     }
 
+    // 0x0046DD06
+    // id : updatingCompanyId global var
+    // payment : ebp
+    // return : ebp == 0x80000000 for false
+    bool ensureCompanyFunding(const CompanyId id, const currency32_t payment)
```
Comment (LeftofZen) [inline src/OpenLoco/src/World/CompanyManager.cpp:624]: what gameplay action triggers this so i can test it?

Comment (duncanspumpkin) [reply src/OpenLoco/src/World/CompanyManager.cpp:624]: All commands that cost something should trigger it. But only ai will trigger the loan bit of it

**Review by AaronVanGeffen — APPROVED** [review]
Logic looks good to me.

**Comment by duncanspumpkin** [comment]
This now has a divergence after my last commit.

**Comment by duncanspumpkin** [comment]
> This now has a divergence after my last commit.

found the issue.

**Pull request merged**